lib/src/components/NxPagination/NxPagination.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import classnames from 'classnames';
+import { min, range, map, curryN } from 'ramda';
+
+import { Props, propTypes } from './types';
+import NxButton from '../NxButton/NxButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { faCaretLeft, faCaretRight } from '@fortawesome/free-solid-svg-icons';
+export { Props, propTypes };
+
+const PAGE_RANGE_SIZE = 5;
+
+function validate(pageCount: number, currentPage: number | null | undefined) {
+  if (pageCount < 0 || !Number.isInteger(pageCount)) {
+    throw new TypeError(`pageCount must be a non-negative integer, was ${pageCount}`);
+  }
+
+  if (currentPage == null) {
+    if (pageCount !== 0) {
+      throw new TypeError(`currentPage must be defined when pageCount is not 0, was ${currentPage}`);
+    }
+  }
+  else {
+    if (pageCount === 0) {
+      throw new TypeError(`currentPage must not be defined when pageCount is 0, was ${currentPage}`);
+    }
+
+    if (currentPage < 0 || currentPage >= pageCount || !Number.isInteger(currentPage)) {
+      throw new TypeError(`currentPage is out of bounds or not an integer: ${currentPage}`);
+    }
+  }
+}
```

Comment:
The following incorporates several suggestions:

* Checks that the values are numeric before checking the range
* Separates checking the pageCount and currentPage into a single condition
* Always prefer `===` to `==` to avoid type coercion errors
* Avoid template strings when not necessary
* Reduce the number of nested conditions by taking advantage of the early return due to throw statements

```suggestion
  if (!Number.isInteger(pageCount) || pageCount <= 0) {
    throw new TypeError(`pageCount must be > 0 if it is defined, was ${pageCount}`);
  }

  if (pageCount && (currentPage === null || currentPage === undefined)) {
    throw new TypeError('currentPage must be defined when pageCount is > 0');
  }

  if (!Number.isInteger(currentPage) || currentPage < 0) {
    throw new TypeError(`currentPage must be a positive integer when the pageCount was defined, was ${currentPage}`);
  }

  if (currentPage >= pageCount) {
    throw new TypeError(`currentPage must be less than the pageCount, currentPage was ${currentPage} and pageCount was ${pageCount}`);
  }
```

Reply:
> The following incorporates several suggestions:

The suggested code doesn't compile.  I also don't necessarily agree with some of the changes:

> Always prefer === to == to avoid type coercion errors

`== null` is explicitly the only time we consider it ok/preferable to use double-equals.  It is the most succinct way to check for both null and undefined.  You'll notice we specifically have our eslint rules set up to allow it.

Reply:
Using `==` for null is fair I suppose - it just seems dangerous to me as it would be easy to miss that when changing the code but as long as we have an eslint rule blocking `==` comparisons against anything other than null I don't mind.

I just quickly wrote that code by hand so I wouldn't be surprised if I missed something syntatically. Maybe I shouldn't have used the suggestion feature but I wanted to make it clear what code would be generally replaced.